File: commonLibrary/src/main/java/com/candy/commonlibrary/utils/LogUtil.java
package com.candy.commonlibrary.utils;

import android.util.Log;

/**
 * Created by zhanghq on 2016/11/30.
 */

public class LogUtil {

    private static boolean isDebug = true;

    public static void disableLog() {
        isDebug = false;
    }

    public static void i(String tag, String msg) {
        if (isDebug) {
            Log.i(tag, msg);
        }
    }

    public static void d(String tag, String msg) {
        if (isDebug) {
            Log.d(tag, msg);
        }
    }

    public static void w(String tag, String msg) {
        if (isDebug) {
            Log.w(tag, msg);
        }
    }

    public static void e(String tag, String msg) {
        if (isDebug) {
            Log.e(tag, msg);
        }
    }
}

File: commonLibrary/src/main/java/com/candy/commonlibrary/utils/LogUtil.kt
package com.candy.commonlibrary.utils

import android.util.Log

/**
 * @Description
 * @Author zhanghaiqiang
 * @Date 2020/9/30 16:20
 */
object LogUtil {
    private var isDebug = true

    fun disableLog() {
        isDebug = false
    }

    fun i(tag: String?, msg: String?) {
        if (isDebug) {
            Log.i(tag, msg)
        }
    }

    fun d(tag: String?, msg: String?) {
        if (isDebug) {
            Log.d(tag, msg)
        }
    }

    fun w(tag: String?, msg: String?) {
        if (isDebug) {
            Log.w(tag, msg)
        }
    }

    fun e(tag: String?, msg: String?) {
        if (isDebug) {
            Log.e(tag, msg)
        }
    }
}